plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -988,6 +1015,13 @@ public boolean supportsLimit()
         return Optional.empty();
     }
 
+    protected Function<String, String> tryApplyLimit(long limit)
+    {
+        return limitFunction()
+                .map(limitFunction -> (Function<String, String>) sql -> limitFunction.apply(sql, limit))
```

Comment:
same here -- 

before the change `tryApplyLimit` was called unconditionally, so it had to accomodate when limit function is not implemented; after the change, the function is called only when limit is supported, and thus

- rename tryApplyLimit -> applyLimit
- replace .map with `.orElseThrow(....)`